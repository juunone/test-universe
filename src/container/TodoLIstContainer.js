import React, { useState } from "react";
import { Input } from "../components/Input";
import { Button } from "../components/Button";
import { List } from "../components/List";
import { deleteOne } from "../utils";

export const TodoListContainer = () => {
  const [state, setState] = useState({ value: "", list: [] });

  const deleteTodoList = key => {
    const res = deleteOne({ list: [...state.list], key });
    setState({
      ...state,
      list: res
    });
  };

  return (
    <div>
      <Input
        value={state.value}
        onChange={e => {
          setState({
            ...state,
            value: e.target.value
          });
        }}
      />
      <Button
        primary
        size="large"
        label="Add"
        backgroundColor="purple"
        color="white"
        onClick={() => {
          const newList = [...state.list];
          if (state.value !== "") newList.push(state.value);
          setState({
            value: "",
            list: newList
          });
        }}
      />
      <List data={state.list} deleteTodoList={deleteTodoList} />
    </div>
  );
};
