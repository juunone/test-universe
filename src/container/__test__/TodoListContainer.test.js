import React from "react";
import { render, fireEvent, screen } from "@testing-library/react";
import { create, act } from "react-test-renderer";

import { TodoListContainer } from "../TodoLIstContainer";

describe("TodoListContainer", () => {
  let root;
  act(() => {
    root = create(<TodoListContainer />);
  });

  it("renders correctly", () => {
    expect(root.toJSON()).toMatchSnapshot();
  });
});
