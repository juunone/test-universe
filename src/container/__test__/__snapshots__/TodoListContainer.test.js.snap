// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoListContainer renders correctly 1`] = `
<div>
  <input
    data-testid="todoInput"
    onChange={[Function]}
    placeholder="투두리스트 입력"
    style={
      Object {
        "color": "black",
      }
    }
    type="text"
    value=""
  />
  <button
    className="storybook-button storybook-button--large storybook-button--primary"
    data-testid="todoButton"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "purple",
        "color": "white",
      }
    }
    type="button"
  >
    Add
  </button>
  <ul />
</div>
`;
